gridfs/synchronous/grid_file.py

```diff
@@ -1472,6 +1474,15 @@ def __init__(
     _file: Any
     _chunk_iter: Any
 
+    def __next__(self) -> bytes:
+        return super().__next__()
+
+    def __next__(self) -> bytes:  # noqa: F811, RUF100
```

Comment:
Any way to avoid the duplicate `def __next__(self)` definitions?

Reply:
Right, but can we workaround that? The duplicate code is strange the read. There's also a runtime perf cost to overriding a method just to call the super class.

Reply:
There isn't a simple way to workaround it, no. We could change the definition to be less confusing, like this:

```
    async def __anext__(self) -> bytes:
        return super().__next__()

    if not _IS_SYNC:

        def __next__(self) -> bytes:  # noqa: F811, RUF100
            raise TypeError(
                "AsyncGridOut does not support synchronous iteration. Use `async for` instead"
            )
```

Which would synchronize to 

```
    def __next__(self) -> bytes:
        return super().__next__()

    if not _IS_SYNC:

        def __next__(self) -> bytes:  # noqa: F811, RUF100
            raise TypeError("GridOut does not support synchronous iteration. Use `for` instead")
```

We can also add a comment explaining why the duplicate def exists.

Reply:
Good idea, but how about:
```python

    if not _IS_SYNC:
        async def __anext__(self) -> bytes:
            return await self.readline()

        def __next__(self) -> bytes:  # noqa: F811, RUF100
            raise TypeError(
                "AsyncGridOut does not support synchronous iteration. Use `async for` instead"
            )
```

Reply:
I forgot we had our own async `readline`, good catch. This looks like we'd read a single line or every byte if the file wasn't line-delimited. Is that the intended behavior for iteration here?

Reply:
Yeah that's what IOBase is supposed to do and AsyncGridOut iteration should match the sync version. We also need to remove IOBase from the async.

Reply:
Done.